tests/Integration/PrestaShopBundle/Controller/Admin/Improve/International/CountryControllerTest.php

```diff
@@ -80,23 +103,72 @@ public function testCreate(int $initialEntityCount): int
      *
      * @param int $countryId
      */
-    public function testEdit(int $countryId): void
+    public function testEdit(int $countryId): int
     {
-        $this->markTestSkipped('Not implemented');
-        // TODO: Implement when edit action is created
+        // this is the default format that is taken when opening country form, if you try to add spaces the test will fail.
+        $addressFormat = 'firstname lastname
+company
+address1 address2
+city, State:name postcode
+Country:name
+phone';
+        //todo: change addressFormat and test if it was correctly changed when address format PR is merged. Now it only takes default value
+        $isoCode = 'BB';
+        $zipCodeFormat = '2NL';
+
+        // First update the country with new data
+        $formData = [
+            'country[name][1]' => 'editName',
+            'country[iso_code]' => $isoCode,
+            'country[call_prefix]' => 1234,
+            'country[default_currency]' => 1,
+            'country[zone]' => 1,
+            'country[need_zip_code]' => '1',
+            'country[zip_code_format]' => $zipCodeFormat,
+            'country[address_format]' => $addressFormat,
+            'country[is_enabled]' => 1,
+            'country[contains_states]' => 0,
+            'country[need_identification_number]' => 0,
+            'country[display_tax_label]' => 1,
+        ];
+        $this->editEntityFromPage(['countryId' => $countryId], $formData);
+
+        // need to clear cache because prestashop caches address format and without clear it sees the one before update
+        Cache::clear();
+
+        // Then check that it was correctly updated
+        $this->assertFormValuesFromPage(
+            ['countryId' => $countryId],
+            $formData
+        );
+
+        $country = new Country($countryId);
+        $this->assertEquals($isoCode, $country->iso_code);
+        $this->assertEquals($zipCodeFormat, $country->zip_code_format);
```

Comment:
This one is overkill since it seems you already checked the form content via `assertFormValuesFromPage` before on L140